packages/nextjs/src/utils/useOnRedirectTo/index.ts

```diff
@@ -10,12 +10,13 @@ const useOnRedirectTo = (
   routes?: Partial<AuthPageRoutes>
 ) => {
   const isAuthRouteRef = useRef<(path: string) => boolean>(() => false);
+  const onRedirectTo = useRef((_path: string, opts?: RedirectOptions) => {});
 
   useMemo(() => {
     isAuthRouteRef.current = (path) => isAuthRoute(path, routes);
   }, [routes]);
 
-  return useCallback((_path: string, opts?: RedirectOptions) => {
+  onRedirectTo.current = (_path: string, opts?: RedirectOptions) => {
```

Comment:
can you explain this change?

Reply:
in order to prevent reinitialize and maintain the latest onRedirectTo reference I used ref instead of useCallback